### DISCUSSION TITLE
Error 401: deleted_client for youtube3-cli

### DISCUSSION THREAD

**Discussion by berezovskyi** [Byron/google-apis-rs#413] [Q&A]
I guess it's connected to https://developers.google.com/identity/protocols/oauth2/resources/loopback-migration#migrate-to-a-supported-alternative

How can I use YT APIs using a command line on a headless machine? Thanks in advance!

**Comment by Byron** [comment]
I don't know, but think it's a question for [yup-oauth2](https://crates.io/crates/yup-oauth2) which deals with auth*.

**Comment by berezovskyi** [comment]
Here is what I figured out so far:

1. On "Create OAuth client ID" in GCP, the _Desktop_ option should be chosen. I tried to outsmart GCP by using _TV and limited input devices_, but no auth mechanisms work with that in practice that would run on the CLI app on a headless machine.
2. An "Installed application flow" (cf. `yup-oauth2`, for example) is used with a local web server (on a remote).
3. A `LocalForward` param is used on the SSH connection to the VM running the code that needs to authorize with Google on my behalf when the auth flow needs to be re-done from time to time.

If someone can help me find a better way, I would be very grateful!
